addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComThermostat3Message.java

```diff
@@ -0,0 +1,182 @@
+import static org.openhab.binding.rfxcom.internal.messages.ByteEnumUtil.fromByte;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComException;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComUnsupportedChannelException;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComUnsupportedValueException;
+
+/**
+ * RFXCOM data class for thermostat3 message.
+ * Mertik G6R-H4x gas heater
+ *
+ * @author Ruud Beukema - Initial contribution
+ */
+public class RFXComThermostat3Message extends RFXComDeviceMessageImpl<RFXComThermostat3Message.SubType> {
+    public enum SubType implements ByteEnumWrapper {
+        MERTIK__G6R_H4T1(0),
+        MERTIK__G6R_H4TB__G6_H4T__G6R_H4T21_Z22(1),
+        MERTIK__G6R_H4TD__G6R_H4T16(2),
+        MERTIK__G6R_H4S_TRANSMIT_ONLY(3);
+
+        private final int subType;
+
+        SubType(int subType) {
+            this.subType = subType;
+        }
+
+        @Override
+        public byte toByte() {
+            return (byte) subType;
+        }
+    }
+
+    /* Operating mode */
+    public enum Commands implements ByteEnumWrapper {
+        OFF(0),
+        ON(1),
+        UP(2),
+        DOWN(3),
+        // For now unimplemented commands
+        RUN_UP__2ND_OFF(4), // G6R_H4T1 and G6R_H4TB respectively only
+        RUN_DOWN__2ND_ON(5), // G6R_H4T1 and G6R_H4TB respectively only
+        STOP(6); // G6R_H4T1 only
+
+        private final int command;
+
+        Commands(int command) {
+            this.command = command;
+        }
+
+        @Override
+        public byte toByte() {
+            return (byte) command;
+        }
+    }
+
+    public SubType subType;
+    public int unitId;
+    public Commands command;
+
+    public RFXComThermostat3Message() {
+        super(PacketType.THERMOSTAT3);
+    }
+
+    public RFXComThermostat3Message(byte[] data) throws RFXComException {
+        encodeMessage(data);
+    }
+
+    @Override
+    public String toString() {
+        String str = "";
+
+        str += super.toString();
+        str += ", Sub type = " + subType;
+        str += ", Device Id = " + getDeviceId();
+        str += ", Command = " + command;
+        str += ", Signal level = " + signalLevel;
+
+        return str;
+    }
+
+    @Override
+    public String getDeviceId() {
+        return String.valueOf(unitId);
+    }
+
+    @Override
+    public void encodeMessage(byte[] data) throws RFXComException {
+        super.encodeMessage(data);
+
+        subType = fromByte(SubType.class, super.subType);
+        unitId = (data[4] & 0xFF) << 16 | (data[5] & 0xFF) << 8 | (data[6] & 0xFF);
+        command = fromByte(Commands.class, (data[7]));
+        // Ignore filler byte at data[8]
```

Comment:
@ruudbeukema according to my version of the SDK there is no filler, are you sure?